spring-rabbit/src/main/java/org/springframework/amqp/rabbit/connection/RabbitConnectionFactoryBean.java

```diff
@@ -629,4 +626,48 @@ protected SSLContext createSSLContext() throws NoSuchAlgorithmException {
 		return SSLContext.getInstance(this.sslAlgorithm);
 	}
 
+
+	/**
+	 * In case the client doesn't provide the sslProperties location or the truststore is null delegate to Java's default
+	 * trust mechanism
+	 * @throws NoSuchAlgorithmException
+	 * @throws KeyManagementException
+	 * @throws KeyStoreException
+     */
+	public void useDefaultTrustStoreMechanism() throws NoSuchAlgorithmException, KeyManagementException,
+			KeyStoreException {
+		SSLContext c = SSLContext.getInstance(this.sslAlgorithm);
+		c.init(null, new TrustManager[] {  getDefaultTrustManager(TrustManagerFactory.getDefaultAlgorithm(), null) }, null);
+		this.connectionFactory.useSslProtocol(c);
+	}
+
+
+	public static X509TrustManager getDefaultTrustManager(String algorithm, KeyStore keystore) throws NoSuchAlgorithmException, KeyStoreException {
+
+		TrustManagerFactory factory;
+		X509TrustManager x509TrustManager = null;
+
+		try {
+			factory = TrustManagerFactory.getInstance(algorithm);
+			factory.init(keystore);
+			for (TrustManager trustManager : factory.getTrustManagers()) {
+
+				if (trustManager instanceof X509TrustManager) {
+					x509TrustManager = (X509TrustManager) trustManager;
+					break;
+				}
+			}
+
+		}
+		catch (NoSuchAlgorithmException e) {
+			throw  e;
+		}
+		catch (KeyStoreException e) {
+			throw  e;
```

Comment:
These catches look redundant.
Although, if you agree with my comment above about `tmf.getTrustManagers()`, we won't need this `getDefaultTrustManager()` method at all.